File: 1779-find-nearest-point-that-has-the-same-x-or-y-coordinate/1779-find-nearest-point-that-has-the-same-x-or-y-coordinate.cpp
class Solution {
public:
    int nearestValidPoint(int x, int y, vector<vector<int>>& points) {
        int dist = INT_MAX;
        int ans = 0;
        for(int i = 0; i < points.size(); i++){
            vector<int> arr = points[i];
            if(arr[0] == x || arr[1] == y){
                int curr = abs(x - arr[0]) + abs(y - arr[1]);
                if(curr < dist){
                    dist = curr;
                    ans = i;
                }
            }
        }
        return dist == INT_MAX ? -1 : ans;
    }
};

File: 1779-find-nearest-point-that-has-the-same-x-or-y-coordinate/1779-find-nearest-point-that-has-the-same-x-or-y-coordinate.java
class Solution {
    public int nearestValidPoint(int x, int y, int[][] points) {
        int dist = Integer.MAX_VALUE;
        int ans = 0;
        for(int i = 0; i < points.length; i++){
            int[] arr = points[i];
            if(arr[0] == x || arr[1] == y){
                int curr = Math.abs(x - arr[0]) + Math.abs(y - arr[1]);
                if(curr < dist){
                    dist = curr;
                    ans = i;
                }
            }
        }
        return dist == Integer.MAX_VALUE ? -1 : ans;
    }
}